js/delivery_receipts.js

```diff
@@ -67,7 +67,7 @@
                         receipt.get('source'),
                         receipt.get('timestamp')
                     );
-                    resolve();
+                    this.remove(receipt);
```

Comment:
Are you sure about this? The point of this in-memory cache is to hold onto early-arriving receipts until the corresponding message arrives, which is often some seconds or milliseconds later in multi-device scenarios.

Reply:
iirc this is most readily observable in the case where messages were sent from another linked device.

Reply:
It's a fair question. The problem I'm seeing is that, over time, more and more unprocessed delivery receipts build up. I'm up to about 280 in one of my installs. They do nothing but make startup slow.

Maybe the right thing to do is to immediately remove them from the retry cache, but keep them in the delivery receipts collection in memory.

Reply:
>Maybe the right thing to do is to immediately remove them from the retry cache, but keep them in the delivery receipts collection in memory.

Yeah, probably best that way. With the current diff I expect many outgoing message will incorrectly appear undelivered until a restart or socket reconnect (which would trigger retrying pending receipts in the cache).

To eliminate potential data loss of in-memory receipts (although I never noticed this happen even before the retry cache was introduced), this receipt collection could be persisted to its own store in indexedDB and reloaded on startup without blocking the loading screen. I think it's ok to be processing delivery receipts while the user is allowed to proceed to the ui, because they don't inject any new message bubbles into the view.